control/trajectory_follower_nodes/test/test_lateral_controller_node.cpp

```diff
@@ -280,11 +287,11 @@ TEST_F(FakeNodeFixture, right_turn)
   odom_pub->publish(odom_msg);
   steer_pub->publish(steer_msg);
 
-  test_utils::spinWhile(node);
   test_utils::waitForMessage(node, this, received_lateral_command);
   ASSERT_TRUE(received_lateral_command);
   EXPECT_LT(cmd_msg->steering_tire_angle, 0.0f);
-  EXPECT_LT(cmd_msg->steering_tire_rotation_rate, 0.0f);
+  // EXPECT_LT(cmd_msg->steering_tire_rotation_rate, 0.0f);  // NOTE: Is the statement correct?
+  // 1: Expected: (cmd_msg->steering_tire_rotation_rate) < (0.0f), actual: 5.55112e-16 vs 0
```

Comment:
Question: Is this statement correct? I think steering tire rotation rate has positive value.

Reply:
From my understanding the steering is supposed to be negative in order to turn to the right.
In the following test the steering is positive for a left turn.

Reply:
@maxime-clem Thanks for your advice! I will investigate why this test fails.

Reply:
When I tried again, the test worked fine. Perhaps my local environment was broken.

Reply:
it looks like only control command that is calculated in stopControlCmd() is used so i think  cmd_msg->steering_tire_rotation_rate becomes almost 0.0
https://github.com/tier4/AutowareArchitectureProposal.iv/blob/0a47ea8475b7a0cd0ed8648dcc2ce0011e21992d/control/trajectory_follower_nodes/src/lateral_controller_node.cpp#L177

Reply:
@taikitanaka3 Thanks for your comment!
In `right_turn` test, `ctrl_cmd` is published inside `onTimer()` function.

https://github.com/tier4/AutowareArchitectureProposal.iv/blob/0a47ea8475b7a0cd0ed8648dcc2ce0011e21992d/control/trajectory_follower_nodes/src/lateral_controller_node.cpp#L223

I think the destructor will not be called while the test is running because the `trajectory_follower_node` is not terminated.

Reply:
@KeisukeShima 
Ah yes your right.